.github/workflows/ci.yaml

```diff
@@ -152,6 +156,7 @@ jobs:
           --cov=xarray
           --cov-report=xml
           --junitxml=pytest.xml
+          --beartype-packages=xarray
```

Comment:
can we move the runtime type-checking to a separate job in `ci-additional`? I don't think we *have* to perform runtime type checking as extensively as the main CI, plus I'd like to have test failures separate from type-checking errors.

Reply:
Note that beartype is supposed to be lightwheight:
> Unbearably fast near-real-time hybrid runtime-static type-checking in pure Python. 

Anyway, I too wouldn't like to mix type checking with other tests for now. The best fit is probably [.github/workflows/ci-additional.yaml](https://github.com/pydata/xarray/blob/main/.github/workflows/ci-additional.yaml) together with PyPy static type checking.